testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java

```diff
@@ -484,6 +484,29 @@ public void testSnappyCompressedParquetTableCreatedInHive()
         onHive().executeQuery("DROP TABLE " + tableName);
     }
 
+    @Test(groups = STORAGE_FORMATS)
+    public void testSnappyCompressedParquetTableCreatedInTrino()
+    {
+        String tableName = "table_trino_parquet_snappy";
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + tableName);
+
+        onTrino().executeQuery(format(
+                "CREATE TABLE %s (" +
+                        "   c_bigint BIGINT," +
+                        "   c_varchar VARCHAR(255))" +
+                        "WITH (format='PARQUET')",
+                tableName));
+
+        onTrino().executeQuery("SET SESSION hive.compression_codec = 'SNAPPY'");
+        onTrino().executeQuery(format("INSERT INTO %s VALUES(1, 'test data')", tableName));
+
+        assertThat(onTrino().executeQuery("SELECT * FROM " + tableName)).containsExactlyInOrder(row(1, "test data"));
```

Comment:
could we somehow add a sanity assertion verifying that written files are actually using snappy compression?

Reply:
I don't know how to do this. I can still improve how the session property is set, to ensure it applies to current catalog.